### PR TITLE
Add NucMorph features from 2023 segmentation dataset

### PR DESCRIPTION
**Pull request by pgarrison**
The currently available NucMorph dataset has some additional features of interest. The colorizer can make them available.

Soon™ we'll have an updated dataset that will (1) rename some of these columns and (2) add new columns, so this will be updated again.

Ran on goldilocks & baby bear, and Peyton tested with colorizer on goldilocks. In the process of running on mama bear.

### CODE REVIEW

File: timelapse-colorizer-data/convert_nucmorph_data.py [hunk 1]
```diff
@@ -81,7 +81,7 @@
 """Column of Y centroid coordinates, in pixels of original image data."""
 OUTLIERS_COLUMN = "is_outlier"
 """Column of outlier status for each object. (true/false)"""
-FEATURE_COLUMNS = ["NUC_shape_volume_lcc", "NUC_position_depth_lcc"]
+FEATURE_COLUMNS = ["NUC_shape_volume_lcc", "NUC_position_depth_lcc", "is_outlier", "edge_cell", "predicted_formation", "predicted_breakdown", "NUC_roundness_surface_area", "NUC_position_height", "NUC_position_width", "termination", "parent_id"]
```
Comment (vianamp) [inline timelapse-colorizer-data/convert_nucmorph_data.py:84]: Not sure `parent_id` is a useful feature here. Thoughts?

Comment (pgarrison) [reply timelapse-colorizer-data/convert_nucmorph_data.py:84]: Not useful for coloring, but possibly useful for filtering?

**Review by ShrimpCryptid — APPROVED** [review]

File: timelapse-colorizer-data/convert_nucmorph_data.py (hunk 1 above)
Comment (vianamp) [reply timelapse-colorizer-data/convert_nucmorph_data.py:84]: Good point!

**Review by vianamp — APPROVED** [review]

**Comment by pgarrison** [comment]
@ShrimpCryptid I had to make some breaking changes to `data_writer_utils` to get this working, but I haven't evaluated their effect on the other conversion scripts. Can you help to get this over the finish line?

**Comment by ShrimpCryptid** [comment]
> @ShrimpCryptid I had to make some breaking changes to `data_writer_utils` to get this working, but I haven't evaluated their effect on the other conversion scripts. Can you help to get this over the finish line?

Unfortunately the changes to `data_writer_utils` breaks the other conversion scripts. Is the data in `termination` a string? 

If you're trying to make a categorical feature, you'll need to do some preprocessing first. I'm realizing it's not documented in the code itself, but [categorical features are all integers under the hood](https://github.com/allen-cell-animated/colorizer-data/blob/main/documentation/DATA_FORMAT.md#6-features). I pushed a commit reverting the changes in `data_writer_utils` and clarifying the `data` array type through hints and commenting.

**Comment by ShrimpCryptid** [comment]
> > @ShrimpCryptid I had to make some breaking changes to `data_writer_utils` to get this working, but I haven't evaluated their effect on the other conversion scripts. Can you help to get this over the finish line?
> 
> Unfortunately the changes to `data_writer_utils` breaks the other conversion scripts. Is the data in `termination` a string?
> 
> If you're trying to make a categorical feature, you'll need to do some preprocessing first. I'm realizing it's not documented in the code itself, but [categorical features are all integers under the hood](https://github.com/allen-cell-animated/colorizer-data/blob/main/documentation/DATA_FORMAT.md#6-features). I pushed a commit reverting the changes in `data_writer_utils` and clarifying the `data` array type through hints and commenting.

And alternatively I can write a method to handle string features, if that was the issue with `termination`!

**Comment by pgarrison** [comment]
> [categorical features are all integers under the hood](https://github.com/allen-cell-animated/colorizer-data/blob/main/documentation/DATA_FORMAT.md#6-features)

Thanks, that explains it! I should have read the docs more closely.

**Review by ShrimpCryptid — APPROVED** [review]
Tested remotely and uploaded to the tool to test, and everything seems to be working. Thank you for helping to raise a couple areas where our documentation could have been clearer!

**Review by toloudis — APPROVED** [review]

**Pull request merged**